### ISSUE TITLE
[BUG] Issues using vBoolean and others for parameters

### ISSUE THREAD

**Issue by tisdall** [collective/icalendar#500]
## Describe the bug
I'm new to using this library so it's possible I'm misunderstanding how these property types are being used.  I was trying to add an `rsvp` parameter to an attendee as a `vBoolean` and got a stacktrace.

The [docs talk about doing it this way](https://icalendar.readthedocs.io/en/latest/usage.html#example):
> Automatic encoding is not yet implemented for parameter values, so you must use the ‘v*’ types you can import from the icalendar package (they’re defined in icalendar.prop):

Looking through the code, it seems like this issue exists for any of the property types that are not based on `str`, `list`, or `tuple`.  `list` and `tuple` go through `q_join()` while everything else goes through `dquote()` which requires it to have a `str.replace()` method.

https://github.com/collective/icalendar/blob/16f9015e16a0376721e66367d72323e7fd676900/src/icalendar/parser.py#L100-L105

## To Reproduce

```python
from icalendar import Event, vBoolean, vCalAddress
attendee = vCalAddress(f'mailto:someone@example.com')
attendee.params['rsvp'] = vBoolean(True)
event = Event()
event.add('attendee', attendee, encode=0)
print(event.to_ical().decode())
```
Output:

(file paths reduced to make it easier to read)
```
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "icalendar/cal.py", line 430, in to_ical
    content_lines = self.content_lines(sorted=sorted)
  File "icalendar/cal.py", line 419, in content_lines
    cl = self.content_line(name, value, sorted=sorted)
  File "icalendar/cal.py", line 412, in content_line
    return Contentline.from_parts(name, params, value, sorted=sorted)
  File "icalendar/parser.py", line 315, in from_parts
    return cls(f'{name}:{values}')
  File "icalendar/parser.py", line 218, in to_ical
    value = param_value(value)
  File "icalendar/parser.py", line 105, in param_value
    return dquote(value)
  File "icalendar/parser.py", line 144, in dquote
    val = val.replace('"', "'")
AttributeError: 'vBoolean' object has no attribute 'replace'
```

## Expected behavior
```
BEGIN:VEVENT
ATTENDEE;RSVP=TRUE:mailto:someone@example.com
END:VEVENT
```

## Environment

 - [x] OS:  MacOS 13.1                              
 - [x] Python version:  3.9.13           
 - [x] `icalendar` version:  5.0.3    

## Additional context

- [ ] I tested it with the latest version `pip3 install https://github.com/collective/icalendar.git`
- [x] I attached the ICS source file or there is no ICS source file

**Comment by jacadzaca** [comment]
Thanks for reporting :) This is definitely not an expected behavior and should be fixed. For now, please use this as a workaround:
```python
from icalendar import Event, vText, vCalAddress
attendee = vCalAddress(f'mailto:someone@example.com')
attendee.params['rsvp'] = vText('TRUE')
event = Event()
event.add('attendee', attendee, encode=0)
print(event.to_ical().decode())
```

Output:
```
BEGIN:VEVENT
ATTENDEE;RSVP=TRUE:mailto:someone@example.com
END:VEVENT
```

**Comment by niccokunzmann** [comment]
Thanks! The fix is in version 5.0.4, ready to install.
